script.speedtester/default.py

```diff
@@ -0,0 +1,939 @@
+        return sum(finished) / (timeit.default_timer() - start)
+
+    def speedtest(self, list=False, mini=None, server=None, share=False, simple=False, src=None, timeout=10,
+    units=('bit', 8), version=False):
+        self.imgPing.setVisible(True)
+        self.imgPing_glow.setVisible(True)
+        startST = ['Executed Speed Test Script']
+
+        global shutdown_event, source
+        shutdown_event = threading.Event()
+
+        socket.setdefaulttimeout(timeout)
+
+        if src:
+            source = src
+            socket.socket = bound_socket
+
+        startST.append('Retrieving speedtest.net configuration')
+        self.update_textbox(startST)
+        try:
+            config = getConfig()
+        except URLError:
+            return False
+
+        startST.append('Retrieving speedtest.net server list')
+        self.update_textbox(startST)
+        self.imgCentertext.setImage(self.image_centertext_testingping)
+
+        servers = closestServers(config['client'])
+
+        startST.append('Testing from %(isp)s (%(ip)s)' % config['client'])
+        self.update_textbox(startST)
+
+        best = getBestServer(servers)
+
+        try:
+            startST.append('Selecting best server based on latency')
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Hosted by: %(sponsor)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Host Server: %(host)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Country: %(country)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('City , State: %(name)s' % best)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            km2mi = 0.62
+            km = '%(d)0.2f ' % best
+            Distance = float(km)
+            miles = Distance * km2mi
+            startST.append('Distance: %s mi' % miles)
+            self.update_textbox(startST)
+        except:pass
+        try:
+            startST.append('Ping: %(latency)s ms' % best)
+            self.update_textbox(startST)
+            self.ping_textbox.setLabel("%.0f" % float(best['latency']))
+        except:pass
+        self.imgCentertext.setImage(' ')
+        self.imgPing.setEnabled(False)
+        self.imgPing_glow.setEnabled(False)
+
+        sizes = [350, 500, 750, 1000, 1500, 2000, 2500, 3000, 3500, 4000]
+        urls = []
+        for size in sizes:
+            for i in range(0, 4):
+                urls.append('%s/random%sx%s.jpg' %
+                                    (os.path.dirname(best['url']), size, size))
+        self.imgGauge.setVisible(True)
+        xbmc.Monitor().waitForAbort(1)
+        self.configGauge(0)
+        self.imgGauge_arrow.setVisible(True)
+
+        startST.append('Testing download speed')
```

Comment:
Hardcoded non localized value

Reply:
Yes, they need to go, but do we have to do that now?
It is English-only now anyway, and the generated image will also be English.